svsim/src/main/scala/vcs/Backend.scala

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package svsim.vcs
+
+import svsim._
+
+object Backend {
+  object CompilationSettings {
+    sealed trait XProp
+    object XProp {
+      case object XMerge extends XProp
+      case object TMerge extends XProp
+    }
+
+    final object TraceSettings {
+      final case class FsdbSettings(verdiHome: String)
+    }
+    final case class TraceSettings(
+      enableVcd:    Boolean = false,
+      enableVpd:    Boolean = false,
+      fsdbSettings: Option[TraceSettings.FsdbSettings] = None) {
+      private def fsdbEnabled = fsdbSettings match {
+        case Some(_) => true
+        case None    => false
+      }
+      private[vcs] def compileFlags = Seq(
+        if (enableVpd || fsdbEnabled) Seq("-debug_acc+pp+dmptf") else Seq(),
+        fsdbSettings match {
+          case None => Seq()
+          case Some(TraceSettings.FsdbSettings(verdiHome)) =>
+            Seq(
+              "-kdb",
+              "-P",
+              s"$verdiHome/share/PLI/VCS/LINUX64/novas.tab",
+              s"$verdiHome/share/PLI/VCS/LINUX64/pli.a"
+            )
+        }
+      ).flatten
+      private[vcs] def verilogPreprocessorDefines = Seq(
+        (enableVcd, svsim.Backend.enableVcdTracingFlag),
+        (enableVpd, svsim.Backend.enableVpdTracingFlag),
+        (fsdbEnabled, svsim.Backend.enableFsdbTracingFlag)
+      ).collect {
+        case (true, value) =>
+          svsim.SvsimCompilationSettings.VerilogPreprocessorDefine(value)
+      }
+      private[vcs] def environment = fsdbSettings match {
+        case None                                        => Seq()
+        case Some(TraceSettings.FsdbSettings(verdiHome)) => Seq("VERDI_HOME" -> verdiHome)
+      }
+    }
+  }
+
+  sealed trait AssertionSettings
+  case class AssertGlobalMaxFailCount(count: Int) extends AssertionSettings
+
+  final case class SimulationSettings(
+    customWorkingDirectory: Option[String] = None,
+    assertionSettings:      Option[AssertionSettings] = None)
+
+  case class CompilationSettings(
+    xProp:                       Option[CompilationSettings.XProp] = None,
+    randomlyInitializeRegisters: Boolean = false,
+    traceSettings:               CompilationSettings.TraceSettings = CompilationSettings.TraceSettings(),
+    simulationSettings:          SimulationSettings = SimulationSettings(),
+    licenceExpireWarningTimeout: Option[Int] = None,
+    archOverride:                Option[String] = None)
+
+  def initializeFromProcessEnvironment() = {
+    (sys.env.get("VCS_HOME"), sys.env.get("LM_LICENSE_FILE")) match {
```

Comment:
OK forgive me on my previous review.
However we shouldn't use `LM_LICENSE_FILE`, that will introduce a huge problem in multi-EDA environment. please use `SNPSLMD_LICENSE_FILE` instead